packages/turf-mask/index.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  Feature,
+  FeatureCollection,
+  Polygon,
+  Position,
+  MultiPolygon,
+} from "geojson";
+import { polygon as createPolygon, multiPolygon } from "@turf/helpers";
+import polygonClipping, { Geom } from "polygon-clipping";
+
+/**
+ * Takes polygons or multipolygons and an optional mask, and returns an exterior
+ * ring polygon with holes.
+ *
+ * @name mask
+ * @param {Polygon|MultiPolygon|Feature<Polygon|MultiPolygon>|FeatureCollection<Polygon|MultiPolygon>} polygon GeoJSON polygon used as interior rings or holes
+ * @param {Polygon|Feature<Polygon>} [mask] GeoJSON polygon used as the exterior ring (if undefined, the world extent is used)
+ * @returns {Feature<Polygon>} Masked Polygon (exterior ring with holes).
+ * @example
+ * const polygon = turf.polygon([[[112, -21], [116, -36], [146, -39], [153, -24], [133, -10], [112, -21]]]);
+ * const mask = turf.polygon([[[90, -55], [170, -55], [170, 10], [90, 10], [90, -55]]]);
+ *
+ * const masked = turf.mask(polygon, mask);
+ *
+ * //addToMap
+ * const addToMap = [masked]
+ */
+function mask<T extends Polygon | MultiPolygon>(
+  polygon: T | Feature<T> | FeatureCollection<T>,
+  mask?: Polygon | Feature<Polygon>
+): Feature<Polygon> {
+  // Define mask
+  const maskPolygon = createMask(mask);
+
+  var polygonOuters = null;
+  if (polygon.type === "FeatureCollection") {
+    polygonOuters = unionFc(polygon);
+  } else if (polygon.type === "Feature") {
+    // Need to cast below as Position[][] isn't quite as strict as Geom, even
+    // though they should be equivalent.
+    polygonOuters = createGeomFromPolygonClippingOutput(
+      polygonClipping.union(polygon.geometry.coordinates as Geom)
+    );
+  } else {
+    // Geometry
+    // Need to cast below as Position[][] isn't quite as strict as Geom, even
+    // though they should be equivalent.
+    polygonOuters = createGeomFromPolygonClippingOutput(
+      polygonClipping.union(polygon.coordinates as Geom)
```

Comment:
Gotta love that type friction

Reply:
In a mind to submit a PR to DefinitelyTyped to tighten up the Position definition. That would hopefully remove the need for concessions like this.

Reply:
I wonder can you use the new  `satisfies` operator here inline before casting to check it matches?

`satisfies Geom as Geom`

Reply:
Gave this a shot and it doesn't seem to work. I think because Position is number[] (possibly with zero elements) and Geom is [number, number] (exactly two elements). Typescript isn't willing to make the call, ultimately barfing with
```
Type 'Position' is not assignable to type 'Pair'.
Target requires 2 element(s) but source may have fewer.
```